rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -1350,20 +1356,20 @@ public void visitConstantExpression(ConstantExpression expression) {
                     jType = JavaType.Primitive.Short;
                 } else if (type == ClassHelper.STRING_TYPE) {
                     jType = JavaType.Primitive.String;
-                    // String literals value returned by getValue()/getText() has already processed sequences like "\\" -> "\"
-                    int length = sourceLengthOfString(expression);
                     // this is an attribute selector
-                    if (source.startsWith("@" + value, cursor)) {
-                        length += 1;
+                    boolean attributeSelector = source.startsWith("@" + value, cursor);
+                    int length = lengthAccordingToAst(expression);
+                    Integer insideParenthesesLevel = getInsideParenthesesLevel(expression);
```

Comment:
How is this even possible? The function is already wrapped in:
```
queue.add(insideParentheses(expression, fmt -> {
```
so I would imagine the parenthesis already being stripped?

Reply:
The `sourceLengthOfString` method previously used here did this under water as well. Notice we're visiting a `ConstantExpression` here, so this doesn't collide with the logic specifically for `MethodCallExpression`s, it merely grabs the value provided by the AST!